### PR TITLE
History per workspace

### PR DESCRIPTION
**Pull request by AntonioCiolino**
Merge for Adding chat history window into the code base.

### CODE REVIEW

File: frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx [hunk 1]
```diff
@@ -140,6 +140,40 @@ export default function WorkspaceSettings({ workspace }) {
...
+                  Chat History
+                </label>
+                <p className="text-xs text-gray-600 dark:text-stone-400">
+                  Chat history: how far back in teh current discussion will 
+                  chat pull responses.
+                  <br />
+                  Default 20. Anything more than 45 is likely to cause significant failures
+                  as well as higher costs.
+                  <br />
+                  Recommended: 20
```
Comment (timothycarambat) [inline frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx:159]: Lets reword this to be more concise
```
The number of previous chats that will be included in the response's short-term memory.
Recommend: 20. Anything more than 45 is likely to lead to continuous chat failures depending on message size.

File: server/models/workspace.js [hunk 2]
```diff
@@ -104,9 +111,10 @@ const Workspace = {
       this.writable.includes(key)
     );
     const values = Object.values(data);
+    // console.log(validKeys);
```
Comment (timothycarambat) [inline server/models/workspace.js:114]: can we remove this commented out console log?

File: server/utils/chats/index.js [hunk 3]
```diff
@@ -104,7 +104,9 @@ async function chatWithWorkspace(workspace, message, chatMode = "chat") {
       error: null,
     };
   } else {
-    const rawHistory = await WorkspaceChats.forWorkspace(workspace.id, 20);
+    var chat_memory = workspace?.openAiHistory;
+    // console.debug(chat_memory);
```
Comment (timothycarambat) [inline server/utils/chats/index.js:108]: can we remove this commented out console log?

Comment (timothycarambat) [reply server/utils/chats/index.js:108]: Also for JS code we utilize `camelCase` naming convention. I also think a name like `messageLimit` would be more indicative of what it is doing as chat memory sound like actual memory objects.

File: server/models/workspace.js [hunk 4]
```diff
@@ -42,6 +44,11 @@ const Workspace = {
                                  END`,
         doif: true,
       },
+      {
+        colName: "openAiHistory",
+        execCmd: `ALTER TABLE ${this.tablename} ADD COLUMN openAiHistory INTEGER DEFAULT 20`,
+        doif: false,
+      },
```
Comment (timothycarambat) [inline server/models/workspace.js:51]: 👍

File: frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx [hunk 5]
```diff
@@ -140,6 +140,40 @@ export default function WorkspaceSettings({ workspace }) {
...
+                </p>
+              </div>
+              <input
+                name="openAiHistory"
+                type="number"
+                min={0.0}
+                max={45.0}
```
Comment (timothycarambat) [inline frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx:166]: These can be whole numbers since step is 1! Not a problem if left though

**Review by timothycarambat — CHANGES_REQUESTED** [review]
Functions well and i tested on my end - just some superficial changes but otherwise looks good once those are down!

File: frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx (hunk 1 above)
Comment (AntonioCiolino) [reply frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx:159]: Changed

File: frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx (hunk 5 above)
Comment (AntonioCiolino) [reply frontend/src/components/Modals/MangeWorkspace/Settings/index.jsx:166]: updated to ints

File: server/models/workspace.js (hunk 2 above)
Comment (AntonioCiolino) [reply server/models/workspace.js:114]: deleted.

File: server/utils/chats/index.js (hunk 3 above)
Comment (AntonioCiolino) [reply server/utils/chats/index.js:108]: camelCased and removed comment
